indigo/indigo/src/main/scala/indigo/shared/formats/GenericTileset.scala

```diff
@@ -0,0 +1,39 @@
+package indigo.shared.formats
+
+import indigo._
+
+/** Utility for working with generic sprite sheets. The
+  * sheet must be uniform, with each sprite having the same
+  * width and height, and a (possibly zero-pixel) margin.
+  */
+final case class GenericTileset(
+    assetName: AssetName,
+    width: Int,
+    height: Int,
+    tileWidth: Int,
+    tileHeight: Int,
+    margin: Int
+) {
+  private val rectangles =
+    for {
+      y <- 0 until height by tileHeight + margin
+      x <- 0 until width by tileWidth + margin
+      rectangle = Rectangle(x, y, tileWidth, tileHeight)
+    } yield rectangle
+
+  protected[formats] def lookupRectangle(idx: Int): Option[Rectangle] =
+    rectangles.lift(idx)
```

Comment:
I don't think this needs to be private/protected. Why would someone wish to look up the rectangle? Who knows! But it's a valid operation on the type and it's self contained so...

Reply:
OK